lib/deep_cover/autoload_tracker.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+
+require 'weakref'
+
+# TODO: if a constant is removed, AutoloadEntries should be removed
+
+module DeepCover
+  class AutoloadTracker
+    AutoloadEntry = Struct.new(:weak_const, :name, :target_path, :interceptor_path) do
+      # If the ref is dead, will return nil, otherwise the target
+      def const
+        weak_const.__getobj__
+      rescue RefError
+        nil
+      end
+    end
+
+    attr_reader :autoloads_by_basename, :interceptor_files_by_path
+    def initialize
+      @autoloads_by_basename = {}
+      @interceptor_files_by_path = {}
+    end
+
+    def autoload_path_for(const, name, path)
```

Comment:
The name `const` is just wrong. It should be `module`. It has to be a `Module` and doesn't have to be a constant at all and `const` is also very easily confused with the constant to be autoloaded...

Reply:
Right, totally forgot, i wanted to go through the whole class to normalize `const` vs `mod`. Thnx

Reply:
Done